File: parking-domain/src/main/java/com/chatchatabc/parking/domain/service/ParkingLotService.java
package com.chatchatabc.parking.domain.service;

import com.chatchatabc.parking.domain.model.ParkingLot;

public interface ParkingLotService {

    /**
     * Save parking lot
     *
     * @param parkingLot the parking lot
     */
    void saveParkingLot(ParkingLot parkingLot);

    /**
     * Verify parking lot
     *
     * @param userId         the user id
     * @param parkingLotUuid the parking lot uuid
     * @return the parking lot
     */
    ParkingLot verifyParkingLot(
            String userId,
            String parkingLotUuid
    ) throws Exception;
}

File: backend/parking-service/src/main/kotlin/com/chatchatabc/parking/domain/service/ParkingLotService.kt
package com.chatchatabc.parking.domain.service

import com.chatchatabc.parking.domain.model.ParkingLot
import org.springframework.stereotype.Service

@Service
interface ParkingLotService {

    /**
     * Register a parking lot
     */
    fun register(ownerId: String, parkingLot: ParkingLot): ParkingLot

    /**
     * Update a parking lot
     */
    fun update(ownerId: String, parkingLotId: String, newParkingLotInfo: ParkingLot): ParkingLot
}